apollo-client/src/main/java/com/ctrip/framework/apollo/spring/spi/DefaultApolloConfigRegistrarHelper.java

```diff
@@ -62,6 +62,10 @@ public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata, B
   }
 
   private String[] resolveNamespaces(String[] namespaces) {
+    // no support for Spring version prior to 3.2.x, see https://github.com/apolloconfig/apollo/issues/4178
+    if (this.environment == null) {
+      return namespaces;
```

Comment:
Before return namespaces, can we use log to warn the user?
```suggestion
      logger.warn("no support for Spring version prior to 3.2.x, if you want use placehoders in the value of @EnableApolloConfig, please upgrade Spring's version to at least 3.3.x or higer version. see https://github.com/apolloconfig/apollo/issues/4178 for more detail");
      return namespaces;
```

Reply:
Or some user not use placeholder, but they use spring version prior to 3.2.x, we can detect the '$', then log a warning.

```suggestion
      if ( namespaces contains '$') {
            logger.warn("xxxxx")
      }
      return namespaces;
```

Reply:
Sounds a good idea.